File: rental-backend/src/main/java/com/realestate/rental/dto/PropertyDto.java
package com.realestate.rental.dto;

import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

// PropertyDto.java
@Data
public class PropertyDto {
    private UUID id;
    private String title;
    private String description;
    private String propertyType;
    private String listingType;
    private BigDecimal price;
    private String currency;
    private String address;
    private String city;
    private String stateProvince;
    private String country;
    private String postalCode;
    private BigDecimal latitude;
    private BigDecimal longitude;
    private Integer bedrooms;
    private Integer bathrooms;
    private BigDecimal areaSqm;
    private Integer floorNumber;
    private Integer totalFloors;
    private Boolean isAvailable;
    private LocalDate availableFrom;
    private String status;
    private Integer viewCount;
    private Integer favoriteCount;
    private List<PropertyImageDto> images;
    private List<PropertyVideoDto> videos;
    private List<AmenityDto> amenities;
    private UserDto landlord;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: rental-app-flutter/lib/features/properties/models/property_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import '../../auth/models/user_dto.dart';
import 'amenity_dto.dart';
import 'property_image_dto.dart';
import 'property_video_dto.dart';

part 'property_dto.freezed.dart';
part 'property_dto.g.dart';

@freezed
abstract class PropertyDto with _$PropertyDto {
  const factory PropertyDto({
    required String id,
    required String title,
    required String description,
    required String propertyType,
    required String listingType,
    required double price,
    required String currency,
    required String address,
    required String city,
    required String stateProvince,
    required String country,
    String? postalCode,
    double? latitude,
    double? longitude,
    int? bedrooms,
    int? bathrooms,
    double? areaSqm,
    int? floorNumber,
    int? totalFloors,
    required bool isAvailable,
    String? availableFrom,
    required String status,
    int? viewCount,
    int? favoriteCount,
    List<PropertyImageDto>? images,
    List<PropertyVideoDto>? videos,
    List<AmenityDto>? amenities,
    UserDto? landlord,
    required String createdAt,
    required String updatedAt,
  }) = _PropertyDto;

  factory PropertyDto.fromJson(Map<String, dynamic> json) => _$PropertyDtoFromJson(json);
}
